src/engraving/libmscore/beam.cpp

```diff
@@ -1031,23 +979,44 @@ void Beam::offsetBeamToRemoveCollisions(const std::vector<ChordRest*> chordRests
         return;
     }
     // tolerance eliminates all possibilities of floating point rounding errors
-    double tolerance = score()->styleMM(Sid::beamWidth).val() * mag() * 0.25 * (_up ? -1 : 1);
+    double tolerance = _beamWidth * 0.25 * (_up ? -1 : 1);
     double startY = (isStartDictator ? dictator : pointer) * spatium() / 4 + tolerance;
     double endY = (isStartDictator ? pointer : dictator) * spatium() / 4 + tolerance;
+    // using <= or >= for doubles is not a very good plan because 0 != 0.000000000001
+    auto fuzzyCompare = [](double a, double b) {
+        double diff = a - b;
+        if (abs(diff) < 0.001) {
+            return 0;
+        } else {
+            return diff < 0 ? -1 : 1;
+        }
+    };
```

Comment:
Maybe you can replace this with `RealIsEqualOrLess` and `RealIsEqualOrMore` from `realfn.h`?

Reply:
Oh, cool, didn't know those were in there. TIL